### ISSUE TITLE
@depends with class: namespace considerations

### ISSUE THREAD

**Issue by iquito** [sebastianbergmann/phpunit#4475]
The newly "upgraded" `@depends` which now supports classes is great, but for me the syntax when it comes to namespaces is a bit surprising:

```php
namespace App\Tests\Integration;

/**
 * This is the syntax that currently has to be used:
 *
 * @depends App\Tests\Integration\OtherTest::class
 */
class SomeTest
{
}

/**
 * Both of these syntax fail (tests are skipped):
 *
 * @depends \App\Tests\Integration\OtherTest::class
 * @depends OtherTest::class
 */
class SomeTest
{
}
```

This seems like an easy pitfall which leads to tests being skipped because of wrong expectations. From my perspective it would be better to handle this more like PHP/docblock annotations:

- Class name contains `\` somewhere: assume it is namespaced from the root namespace, and allow a `\` at the beginning even though it is not necessary
- Class name does not contain `\`: assume it is in the same namespace as the current test. This will be a common case and make the `@depends` more readable, because only the class name without namespace has to be included.

This would be almost consistent with PHP and annotations without breaking anything. The only edge case would be if somebody uses `@depends OtherTest::class` and OtherTest has no namespace (is in the global namespace), yet that could easily be solved by writing `@depends \OtherTest::class`. The depends-on-class feature is also very new and is not documented yet, so making the syntax more similar to PHP/annotations now before it is widely used might prevent quite a bit of confusion.

**Comment by iquito** [comment]
One more small reason: PHPStorm also marks the current way of qualifying the namespace with "not found", because without the leading `\` it assumes it is a relative namespace.

**Comment by sebastianbergmann** [comment]
I am looking forward to a time when we do not need to care about [names in comments anymore](https://github.com/sebastianbergmann/phpunit/issues/4506).

**Comment by jtheuerkauf** [comment]
@sebastianbergmann i agree with you, but i think it's going to be a long time before every project is off pre-attribute versions of PHP. In the meantime, i hope there will be some fix for this. Since it functions as long as the root `\` for the namespace is omitted, i can understand why this is low priority.

i've also registered an issue with JetBrains because the inspection UX gets weird from this. 
https://youtrack.jetbrains.com/issue/WI-66961

PS: Thank you for everything.
